### PR TITLE
OSASINFRA-3571: openstack: refactor ConfigDrive & add it for workers

### PR DESCRIPTION
**Pull request by EmilienM**
* Create a function to figure out whether IPv6 single stack is enabled
  and that will activate ConfigDrive.
* Enable ConfigDrive for workers as well, not just masters.

### CODE REVIEW

File: pkg/asset/machines/openstack/machines.go [hunk 1]
```diff
@@ -161,7 +166,7 @@ func Machines(ctx context.Context, clusterID string, config *types.InstallConfig
 	return machines, controlPlaneMachineSet, nil
 }
 
-func generateProviderSpec(ctx context.Context, clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, role, userDataSecret string, failureDomain machinev1.OpenStackFailureDomain) (*machinev1alpha1.OpenstackProviderSpec, error) {
+func generateProviderSpec(ctx context.Context, clusterID string, platform *openstack.Platform, mpool *openstack.MachinePool, osImage string, role, userDataSecret string, failureDomain machinev1.OpenStackFailureDomain, configDrive *bool) (*machinev1alpha1.OpenstackProviderSpec, error) {
```
[the remark this answers is not in the document]
Comment (EmilienM) [reply pkg/asset/machines/openstack/machines.go:169]: I need coffee.

File: pkg/asset/machines/openstack/machinesets.go [hunk 2]
```diff
@@ -34,6 +34,9 @@ func MachineSets(ctx context.Context, clusterID string, config *types.InstallCon
 	}
 	mpool := pool.Platform.OpenStack
 
+	// Only enable config drive when using single stack IPv6
+	configDrive := isSingleStackIPv6(config.MachineNetwork)
```
[the remark this answers is not in the document]
Comment (2uasimojo) [reply pkg/asset/machines/openstack/machinesets.go:38]: Disregard. I see now that `Networking` is inlined. Thought I was going crazy.

**Pull request merged**